<!--/*****************************************************************
 * labust_mission_test.launch
 *
 *  Created on: Apr 15, 2014
 *      Author: Filip Mandic
 *
 ********************************************************************/

/*********************************************************************
* Software License Agreement (BSD License)
*
*  Copyright (c) 2014, LABUST, UNIZG-FER
*  All rights reserved.
*
*  Redistribution and use in source and binary forms, with or without
*  modification, are permitted provided that the following conditions
*  are met:
*
*   * Redistributions of source code must retain the above copyright
*     notice, this list of conditions and the following disclaimer.
*   * Redistributions in binary form must reproduce the above
*     copyright notice, this list of conditions and the following
*     disclaimer in the documentation and/or other materials provided
*     with the distribution.
*   * Neither the name of the LABUST nor the names of its
*     contributors may be used to endorse or promote products derived
*     from this software without specific prior written permission.
*
*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
*  POSSIBILITY OF SUCH DAMAGE.
******************************************************************/-->

<launch>
    <!-- ========================================================= -->
    <!-- == Launch settings ====================================== -->
    <!-- ========================================================= -->

    <!-- Mission execution settings ============================== -->
    
    <arg name="labust_mission_path" value ="$(find labust_mission)" /> 
    <arg name="xml_save_path" value ="$(find labust_mission)/data/mission.xml" /> 
    <arg name="mission_execution_INTEGRATED" default="false" />
    <arg name="mission_execution" default="true" />
    <arg name="mission_parser" default="true" />
    <arg name="neptus_parser" default="true" />
    <arg name="parseGUI" default="true" />
   
    <!-- Simulation settings ===================================== -->  

    <arg name="is_sim" value="$(optenv IS_SIM 0)" />
    <rosparam command="load" file="$(find snippets)/data/locations/$(env LOCATION).yaml" />

    <!-- Control settings ======================================== -->

    <arg name="use_ident" value="$(optenv USE_IDENTIFICATION 0)" />

    <!-- Navigation settings ======================================== -->

    <arg name="ekfAbs" default="true" />

    <!-- Vehicle settings ======================================== -->

    <arg name="vehicle" value ="pladypos" /> 
    <rosparam command="load" file="$(env MODEL)" />


    <!-- ========================================================= -->
    <!-- == Launch includes ====================================== -->
    <!-- ========================================================= -->

    <!-- labust_sim include ====================================== -->

    <include
        if="$(arg is_sim)" 
        file="$(find labust_mission)/launch/labust_sim_template.launch">
    </include>

    <!-- labust_navigation include =============================== -->

    <include file="$(find labust_mission)/launch/labust_navigation_template.launch">
        <arg name="ekfAbs" value="$(arg ekfAbs)" />
    </include>

    <!-- labust_control include ================================== -->

    <include file="$(find labust_mission)/launch/labust_control_template.launch">
       <arg name="use_ident" value="$(arg use_ident)" />
    </include>

    <!-- labust_primitives include =============================== -->

    <include file="$(find labust_mission)/launch/labust_primitives_template.launch"> </include>

    <!-- labust_mission include ================================== -->

    <include file="$(find labust_mission)/launch/labust_mission_template.launch"> 
 
        <arg name="labust_mission_path" value ="$(arg labust_mission_path)/" /> 
        <arg name="xml_save_path" value ="$(arg xml_save_path)" /> 
        <arg name="mission_execution" value="$(arg mission_execution)" /> 
        <arg name="mission_execution_INTEGRATED" value="$(arg mission_execution_INTEGRATED)" /> 
        <arg name="mission_parser" value="$(arg mission_parser)" />
        <arg name="neptus_parser" value="$(arg neptus_parser)" />
        <arg name="parseGUI" value="$(arg parseGUI)" />

    </include>

    <!-- labust_visualization include ============================ -->

    <include file="$(find labust_mission)/launch/labust_visualization_template.launch"> </include>

</launch>
